### PR TITLE
Implement Decorator Pattern for Correcting Names

### PR DESCRIPTION
**Pull request by henask12**
## Pull Request: Implement Decorator Pattern for Correcting Names

**Description:**
This pull request implements the Decorator design pattern in Ruby to correct the names of people, following the project requirements. It introduces a `Nameable` class as an interface and extends the existing `Person` class to inherit from it. Additionally, it creates three decorator classes: `Decorator`, `CapitalizeDecorator`, and `TrimmerDecorator`, to modify the behavior of the `correct_name` method.

**Changes Made:**
- Created a `Nameable` class with an abstract method `correct_name`.
- Modified the existing `Person` class to inherit from `Nameable` and implemented the `correct_name` method to return the person's name.
- Created a base decorator class `Decorator` that inherits from `Nameable` and accepts a `nameable` object as a parameter in its constructor.
- Created a `CapitalizeDecorator` class that capitalizes the output of `@nameable.correct_name`.
- Created a `TrimmerDecorator` class that ensures the output of `@nameable.correct_name` has a maximum of 10 characters and trims it if it's longer.

**Testing:**
- Created instances of the `Person`, `CapitalizeDecorator`, and `TrimmerDecorator` classes to demonstrate how decorators modify the behavior of the `correct_name` method.
- Tested and verified that the decorators correctly capitalize and trim names as expected.

**Checklist:**
- [x] Implemented the Decorator pattern for name correction.
- [x] Modified the `Person` class to inherit from `Nameable` and implemented the `correct_name` method.
- [x] Created base `Decorator` class and concrete decorator classes (`CapitalizeDecorator`, `TrimmerDecorator`).
- [x] Tested the decorators with sample data to ensure correct behavior.
- [x] Ensured compliance with Ruby best practices.
- [x] Documented the code following professional standards.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class CapitalizeDecorator < Decorator
+    def correct_name
+      super.capitalize
+    end
+  end
+
```
Comment (V-Blaze) [inline capitalize_decorator.rb]: - You have done great so far :+1: But there are a few changes you need to make before moving to the next milestone. :smile: :wave: 

- [ ] According to the project requirement, you are meant to run this test to see your decorator in action.

```sh
person = Person.new(22, 'maximilianus')
person.correct_name
capitalized_person = CapitalizeDecorator.new(person)
puts capitalized_person.correct_name
capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
puts capitalized_trimmed_person.correct_name
```
-  [ ] When I ran these, I got an error in the console, I'll suggest you go through your classes and make sure everything is implemented correctly, and then run the code to make sure the result you get is what's expected. :smile: :small_airplane: :wave: 

- - Screenshot for reference:
Screenshot from 2023-09-05 09-38-41

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @henask12 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- All required classes created ✔
-  Some class methods implemented correctly 👋 
-  Correct git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+require_relative 'decorator'
+require_relative 'nameable'
+require_relative 'trimmer_decorator'
+require_relative 'capitalize_decorator'
+
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
   def initialize(name, age = 'Unknown', parent_permission: true)
```
Comment (julie-ify) [inline person.rb:10]: - [ ] **[OPTIONAL]** From a previous project this should be

  ```
  age, name = 'Unknown', parent_permission: true
  ```

**Review by julie-ify — APPROVED** [review]
Hi @henask12,

## Highlights!
- Good code organization and working decorator ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**